lib/utils/dom.js

```diff
@@ -188,31 +188,33 @@ export function scrollTo(x, y) {
 	window.scrollTo(x, y - headerOffset);
 }
 
-export const getHeaderOffset = _.once(() => {
-	let header;
+export const getHeaderOffset = _.memoize((includePartialWidthHeader = false) => {
+	let headers = [];
 
 	// ideally we would import betteReddit directly,
 	// but we can't depend on any modules without causing a bird's nest of circular dependencies
 	if (Modules.isRunning('betteReddit')) {
 		switch (Modules.get('betteReddit').options.pinHeader.value) {
+			case 'userbar':
+				if (includePartialWidthHeader) header.push(document.getElementById('header-bottom-right'));
+				break;
 			case 'sub':
+				headers.push(document.getElementById('sr-header-area'));
+				break;
 			case 'subanduser':
-				header = document.getElementById('sr-header-area');
+				headers.push(document.getElementById('sr-header-area'));
+				if (includePartialWidthHeader) headers.push(document.getElementById('header-bottom-right'));
 				break;
 			case 'header':
-				header = document.getElementById('header');
+				headers.push(document.getElementById('header'));
 				break;
 			case 'none':
 			default:
 				break;
 		}
 	}
 
-	if (header) {
-		return header.offsetHeight + 6;
```

Comment:
I couldn't find a reason for adding `6` if any header is pinned, so I removed it.